pkg/controller/gitopsconfig/controller.go

```diff
@@ -75,8 +76,22 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// sigs.k8s.io/controller-runtime v0.1.10 is used now
+	// If the version were at least 0.2.0, a ready predicate.GenerationChangedPredicate could be used instead of writing this one
+	// https://godoc.org/sigs.k8s.io/controller-runtime/pkg/predicate#GenerationChangedPredicate
+	generationChangedPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			// If there's a status update, .metadata.Generation field isn't changed - ignore such event
+			return e.MetaOld.GetGeneration() != e.MetaNew.GetGeneration()
```

Comment:
Shouldn't we also check if e.MetaOld != nil and e.MetaNew != nil, as is done in the original GenerationChangedPredicate you linked to?

Reply:
I thought it a rather unlikely case but better to be safe than sorry.